bip-newproposal.mediawiki

```diff
@@ -0,0 +1,140 @@
+
+===4. Bitcoin Script Integration===
+* '''Taproot Leaf Extension'''  
+<syntaxhighlight lang="text">
+OP_1 <pubkey_combined> OP_CHECKSIG_HYBRID
+</syntaxhighlight>
+
+: <code>pubkey_combined = (pk_dil, pk_sch)</code>  
+: <code>OP_CHECKSIG_HYBRID</code>: verifies Schnorr on-chain, binds Dilithium off-chain.
+
+* '''Soft Fork Opcode'''  
+Introduce <code>OP_CHECKSIG_QES2</code>:  
+<syntaxhighlight lang="text">
+Stack: [signature, pubkey]
+Parse σ = ((c,z),(R,s)), pk = (pk_dil, pk_sch)
+h = H2(c||message)
+result = Schnorr.Verify(pk_sch, h, (R,s))
+Push result
+</syntaxhighlight>
+
+===5. Transaction Format===
+* Public key: 33 bytes (unchanged).  
+* Schnorr signature: 64 bytes (unchanged).  
+* Witness: includes Dilithium challenge vector <code>c</code>.  
+* Off-chain storage: full Dilithium <code>(c, z)</code>.  
+
+==Rationale==
+
+* '''Efficiency''': Only Schnorr (64 bytes, O(λ) ops) validated on-chain.  
+* '''Security''': Dilithium ensures quantum resistance; Schnorr ensures compatibility.  
+* '''Modularity''': Full verification optional for archive/Tier-2 nodes.  
+* '''Backward compatibility''': Transactions indistinguishable from Schnorr-only.
+
+==Comparison: QES2_ECDSA vs QES2_Schnorr==
+
+Early versions of QES2 proposed a hybrid with ECDSA. However:
+* ECDSA lacks native Taproot support and adds legacy script complexity.
+* Schnorr (BIP340) integrates seamlessly with Taproot.  
+* QES2_Schnorr eliminates the need to expose public keys directly in the script path.  
+* Key-path spend is intentionally disabled to ensure that PQC guarantees are always enforced via script path commitments.
+
+Thus, QES2_Schnorr is preferred as the long-term solution.
+
+==Taproot Script Path Considerations==
+
+* PQ-Derived Schnorr is integrated via Taproot leaves only.  
+* Key-path spend is disabled to avoid bypassing Dilithium binding.  
+* This ensures that all spends must go through the hybrid verification path, preserving post-quantum guarantees.  
+
+==Relation to P2QRH==
+
+P2QRH (Pay-to-Quantum-Resistant-Hash) has been proposed as a generic quantum-safe output format.  
+PQ-Derived Schnorr is compatible and potentially complementary:
+* P2QRH provides quantum-safe key commitment.  
+* PQ-Derived Schnorr provides quantum-safe transaction signatures.  
+* Together, they can form a layered defense against quantum adversaries.  
+
+==Backward Compatibility==
+
+* Legacy nodes interpret PQ-Derived Schnorr transactions as standard Schnorr.  
+* Dilithium verification is optional off-chain and does not affect consensus.  
+* New opcode <code>OP_CHECKSIG_QES2</code> can be activated via soft fork.  
+
+==Security Considerations==
+
+* Schnorr-only reliance collapses under quantum adversaries.  
+* Full Dilithium verification is necessary for PQC security.  
+* Binding <code>h = H2(c || m)</code> prevents mixing Schnorr and Dilithium signatures.  
+* Off-chain storage redundancy (e.g., IPFS, Arweave) is required for Dilithium proofs.  
```

Comment:
Given the criticality of the full signature data being available for transaction verification, I would expect this point to be further elaborated for this proposal to be considered technically sound.